### PR TITLE
ISPO Working Group Charter

### PR DESCRIPTION
**Pull request by rrrutledge**
View the [rendered markdown](https://github.com/InnerSourceCommons/foundation-governance/blob/rrrutledge-patch-1/ispo-working-group-charter.md) and [example project board](https://github.com/orgs/InnerSourceCommons/projects/3/views/1).

### CODE REVIEW

**Review by zkoppert — APPROVED** [review]
Looks great @rrrutledge! I like the balanced approach of everybody is special **and** much is sharable. 🎉

File: ispo-working-group-charter.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+The purpose of this working group is to support ISPOs with materials needed to teach and scale InnerSource in their respective company.
+This material could be directed towards senior leaders, managers, or individuals in a variety of roles.
+
+# Summary
+
+We believe that most training and teaching content can be shared across companies when those institutions share the same relevant points of context.
+While not every company is exactly the same, not every company is wholly different, either.
+For each challenge in InnerSource, we believe that there may be about a half-dozen ways of solving it (depending on context) that, as a set, work for 80% of companies out there.
```
Comment (spier) [inline ispo-working-group-charter.md:8]: This sentence was pretty hard to understand for me.

**Review by spier — COMMENTED** [review]
Great work @rrrutledge! 

I love the idea of setting a charter before starting a Working Group.
Reminds me of the RFC process, which I happened to read something about last week.

I left suggestions for minor fixes inline in the PR.

Some general thoughts:

**1) "what" over "how"**

I would try to focus as much as possible on the "what" and even less on the "how". e.g. I am wondering if the weekly meeting and the Kanban board are too much info, if those should rather be worked out by the members of the WG itself once it kicks off?

**2) membership**

Who would be the most desirable members of the WG? Or what different types of people do you need?

Sounds like you need
a) people already working in ISPOs/OSPOs or similar that are willing to share what they learned
b) people in the process of starting an ISPO or similar

While this point sounds like talking about the "how" as well, it might influence what the WG would work on quite a lot. So finding a good mix in the beginning would be key.

**3) Alternatives**

What are the alternatives to starting this WG? 
You probably have most of this info in your head already. Writing it down will highlight the pros/cons of the different approaches and increase clarity about why you picked the approach of starting the WG.

Alternatives I can think of are:

- do nothing
- inject these topics into the existing working groups

Read more about the importance of considering competition/alternatives in here:
https://philcalcado.com/2018/11/19/a_structured_rfc_process.html

4) Ties to other WGs

> It is expected that the implementation of many solutions may be contributions to other working groups in the InnerSource Commons (e.g. Patterns, Learning Path).

It was happy to see this note. Could be worthwhile to spell this out a bit further, if you already have some thoughts about the desired relationships between the WGs. 

5) Sample Code

>  These implementations will likely be documents, videos, **or code**, each of which is generic enough to be used across companies

The bit about "code" here seems new, when compared to the other WGs, which are exclusively about collecting docs (Patterns WG) and docs+videos (Learning Path).

If these sample implementations are key for this new WG, I would try to highlight it more.

Again, great work.
Looking forward to see what this will lead to 🚀

File: ispo-working-group-charter.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+These two examples follow the same pattern - providing directly-usable materials that help ISPOs to advance their mission in their company.
+
+# Implementation
+
+This working group will track the production of solutions to challenges with InnerSource.
+We will track this production on a Kanban board with the following columns:
+
+1. **Challenge**.  This means that we've identified such-and-such a thing as a challenge.
+1. **Proposed**.  We have a proposal for a solution to the challenge.
+1. **Internal**.  We have implemented the solution internal in a company.
```
Comment (fioddor) [inline ispo-working-group-charter.md:33]:
```suggestion
1. **Internal**.  We have implemented the solution internally in a company.
```

**Comment by rrrutledge** [comment]
Thoughts from @claredillon :
* This is related to search for content.
* Some things (e.g. FAQ) may not have an existing working group location (e.g. patterns, learning-path).
* What is the license?  CC-BY-SA may be tough, here.  CC-BY.
* How can the foundation help the process.  It is easy to start things, but difficult to finish.  How can we encourage things to finish without many one-on-one relationships.

File: ispo-working-group-charter.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+1. **External**.  We have implemented the solution externally.
```
[the remark this answers is not in the document]
Comment (dicortazar) [reply ispo-working-group-charter.md:34]: I assume this is for consultancy companies that can share expertise or their own method?

**Comment by claredillon** [comment]
Overall, just want to say what a marvelous initiative - so happy to see it here and would be delighted to participate to help feed in asks as we get them and perhaps support the creation of the end result.

**Comment by claredillon** [comment]
Some more details on the thoughts above:
- Regarding the license - it's an observation that we publish most things in ISC under CC-BY-SA - but that requires any modified versions to be shared under the same license. This can be a disincentive to individuals in corporations using our material to begin with (because they may be uncertain they _can_ share modified versions if there are any internal corporate info on them, even a logo). Perhaps for these materials we might look to CC-BY to incentivise re-use?? The risk is we may not ever see the end result - but we wouldn't anyhow if a CC-BY-SA license is constraining the desired behaviour... I do think a decision on this would be helpful upfront as it may influence how/if folks contribute and how impactful the end result is.

**Comment by claredillon** [comment]
- Another question I would have is - what happens if someone comes along with a fully formed asset/learning resource to donate. Is there a process for quality checking content? Is there any rebranding that is considered necessary (e.g. to include ISC logo on the published result??). Decisions on this not necessary to get started - just curious as to thoughts about a good approach here.

**Comment by claredillon** [comment]
- On support - the observation made to Russ is that people can often be up for starting things like pattern write-ups / articles - but some formal editorial / design support can often help get the final version over the line. Wondering what might help here. Let's see and perhaps we can support through Foundation. We have access to formal editorial services for example. Doesn't have to be decided to get started - just wanted to raise the option for discussion if needed later.
- I'll also note that we have had interest from some organizations in sponsoring the creation of specific materials for them to put their brand on (instead of being interested in becoming a long term supporter for example). This sometimes ties more closely to their corporate goals. So this could be a way some of this gets funded. 


(sorry about all the comments - I had this in one big comment - but thought it might be useful to split it into topics)

**Comment by claredillon** [comment]
> * Russ mentions the FAQ above - we've been talking for ages in the Marketing WG about the need for a "Golden Deck" and more recently about having some FAQs which address the most commonly asked questions that we get when people join the Slack. It's been really challenging to get to a finish point on the Golden Deck. If this group could help this, that would be great.
> * Another point here is that things like a set of FAQs and answers could help serve as a guide into our content. For example, someone comes and asks - how do I get started? And we often point to the patterns or the learning path WG as a whole - but that can be too much to consume as a starting point, I think what people are often looking for are snippets that help them contextualize those assets. This group's work could really help that. It may not have a place under the existing Patterns or LP - but we'd find a place on the website for them!!

Mini pieces of content that relate to a specific question e.g. "How do I gamify InnerSource participation?" can also be excellent material for social reach. A little video / article just looking at answering just that question can get much more traction than "read the book" or even "watch that talk" type answers.

File: ispo-working-group-charter.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+As another example, to the challenge of `How do I show InnerSource KPIs?`,
+this working group could ensure that there is a software project that will produce charts of InnerSource activity from sources like _GitHub_ and Jira.
+The project could be configured in a (finite) number of ways, depending on varying setup of _GitHub_ and Jira that affects tha report views.
+
+These two examples follow the same pattern - providing directly-usable materials that help ISPOs to advance their mission in their company.
+
+# Implementation
+
+This working group will track the production of solutions to challenges with InnerSource.
```
Comment (gruetter) [inline ispo-working-group-charter.md:28]: Is it "only" about tracking or rather about _driving discovery and formalization_ of soutions, @rrrutledge ?

File: ispo-working-group-charter.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (gruetter) [reply ispo-working-group-charter.md:34]: Or is it perhaps simply about what we call _Known Instances_ in our patterns? Meaning: it's a solution which has proven successful in more than one company?

File: ispo-working-group-charter.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+We will track this production on a Kanban board with the following columns:
+
+1. **Challenge**.  This means that we've identified such-and-such a thing as a challenge.
+1. **Proposed**.  We have a proposal for a solution to the challenge.
+1. **Internal**.  We have implemented the solution internally in a company.
+1. **External**.  We have implemented the solution externally.
+1. **Validated**.  More than one company is using the solution.
+
+We will have a weekly meeting of all ISPO owners where we will list new challenges, discuss proposed solutions, and demo finished solutions.
```
Comment (gruetter) [inline ispo-working-group-charter.md:37]: Do we want to exclude people who don't "own" ISPOs?

**Comment by rrrutledge** [comment]
Thanks for all of these comments!  I will take a look soon and make some updates.

**Comment by rrrutledge** [comment]
Responding to @spier:

> if the weekly meeting and the Kanban board are too much info

I think that your right _in theory_, but in practice we've seen this pattern work multiple times across multiple groups.  I think that it would be a disservice to make the working group figure this out on their own instead of just starting them with this best practice.

> So finding a good mix in the beginning would be key.

Good point.

> Alternatives I can think of are:
> 
> * do nothing
> * inject these topics into the existing working groups

"Do nothing" is our current strategy.  I'm proposing this working group because I think it will be better than doing nothing.
At the end of the charter I call out that most of the final output of this working group will go into other working groups.

> already have some thoughts about the desired relationships between the WGs

Don't have any more specific thoughts.  Just anticipate that our "solution"s often times will be to make a contribution to another working group.

> If these sample implementations are key

I don't know what will be key.  We'll have to see what develops as the working group gets going. Just trying to outline some of the possibilities.

> Looking forward to see what this will lead to 🚀

Thanks!  Me, too.

**Comment by rrrutledge** [comment]
Responding to @claredillon:

> what a marvelous initiative

Thanks!  You use adjectives that make people feel special (... `brilliant`!).  Thank you!

> we might look to CC-BY to incentivise re-use

Good point.

> what happens if someone comes along with a fully formed asset/learning resource

We need things to be usable across companies in order to reach the **External** state, meaning that they don't contain company-specific instructions.  I don't think that company-specific branding is necessarily a blocker for reaching the **External** state.  It may even motivate people to contribute if they know that their company can get credit in the industry for owning a widely-used piece of training content (Spotify Guild model, anyone?).

> the need for a "Golden Deck"

This would be the kind of thing that this ISPO working group would want to track production of.  People are making these kinds of things, they're just not sharing them.  What I hope is the niche of this working group is just making plain this situation through the **Internal** and **External** states on our project board.  Visibility, language, and structure are powerful motivators and catalysts for action, and this working group provides all 3 to this problem of great content staying internal to companies.

> people are often looking for are snippets that help them contextualize those assets. This group's work could really help that.

Yes!  There is a gap where other working groups lead with the asset-first whereas this working group leads with challenge first.  I think that the asset-first approach makes it harder for people to find what they need or even know what they need and so the working groups don't receive as much contribution or engagement.  I am hoping that this ISPO working group can have the dual role of helping people to find what they need _and_ helping working groups get more contributors.  A double whammy.

> We have access to formal editorial services for example

This could be great.  Fantastic content is often the result of collaboration between a content expert and a communication expert (ghost writers, speech writers, documentary creators, etc).

> sponsoring the creation of specific materials for them to put their brand on

This is fantastic.  As mentioned before, I think that this approach is a great motivator for people to spend time contributing.

**Review by rrrutledge — COMMENTED** [review]
Thanks for the feedback!

**Comment by rrrutledge** [comment]
Thanks, everyone!  This pull request may be the one of mine that has received the most amount of comments, ever.  Thank you for this collaboration and feedback.

File: ispo-working-group-charter.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+The purpose of this working group is to support InnerSource Program Offices (ISPOs) with materials needed to establish themselves, teach and scale InnerSource in their respective company.
```
Comment (spier) [inline ispo-working-group-charter.md:1]:
```suggestion
# ISPO Working Group Charter

The purpose of this working group is to support InnerSource Program Offices (ISPOs) with materials needed to establish themselves, teach and scale InnerSource in their respective company.
```

File: ispo-working-group-charter.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+This material could be directed towards senior leaders, managers, or individuals in a variety of roles.
+
+# Summary
```
Comment (spier) [inline ispo-working-group-charter.md:4]:
```suggestion
## Summary
```

File: ispo-working-group-charter.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+We believe that most training and teaching content can be shared across companies when those institutions share the same relevant points of context.
+While not every company is exactly the same, not every company is wholly different, either.
+For each challenge in InnerSource, we believe that there are a small number of ways of common ways to address it that should work for most companies.
+Our goal is to document these sets of solutions to challenges, make it easy for ISPOs to find which apply to them, and then to produce sharable, open source implementations of these solutions.
+These implementations will likely be documents, videos, or code, each of which is generic enough to be used across companies,
+yet specific enough that people in the company can leverage them to solve their problems without needing notable amounts of company-specific work.
+
+# Examples
```
Comment (spier) [inline ispo-working-group-charter.md:13]:
```suggestion
## Examples
```

File: ispo-working-group-charter.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+There would be different clips that highlight different benefits of InnerSource,
+and the ISPO could select those videos that highlight the benefits that their Senior Leaders care about.
+
+As another example, to the challenge of `How do I show InnerSource KPIs?`,
+this working group could ensure that there is a software project that will produce charts of InnerSource activity from sources like _GitHub_ and Jira.
+The project could be configured in a (finite) number of ways, depending on varying setup of _GitHub_ and Jira that affects tha report views.
+
+These two examples follow the same pattern - providing directly-usable materials that help ISPOs to advance their mission in their company.
+
+# Implementation
```
Comment (spier) [inline ispo-working-group-charter.md:26]:
```suggestion
## Implementation
```

File: ispo-working-group-charter.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+# Examples
+
+As an example, to the challenge of `How do I sell InnerSource to Senior Leadership?`,
+this working group could ensure that there are video clips explaining various benefits of InnerSource from a Senior Leader perspective.
+There would be different clips that highlight different benefits of InnerSource,
+and the ISPO could select those videos that highlight the benefits that their Senior Leaders care about.
+
+As another example, to the challenge of `How do I show InnerSource KPIs?`,
+this working group could ensure that there is a software project that will produce charts of InnerSource activity from sources like _GitHub_ and Jira.
+The project could be configured in a (finite) number of ways, depending on varying setup of _GitHub_ and Jira that affects tha report views.
```
Comment (spier) [inline ispo-working-group-charter.md:22]: Just curious why "GitHub" is in italics, while "Jira" isn't?

**Review by spier — COMMENTED** [review]
Left some more comments about the formatting of the headlines :)

Key change is that I added a H1 title to the top of the doc, and moved all other titles to H2.

**Comment by souravcipher** [comment]
ISPO WG looking exciting!! Love to contribute.

**Comment by rrrutledge** [comment]
Glad you’re looking forward to it, @souravcipher ❗️ Excited to discuss more at next weeks community call!

**Comment by spier** [comment]
Details of the mentioned community call in here:
https://www.eventbrite.com/e/an-ispo-working-group-for-the-isc-tickets-470944857747

File: ispo-working-group-charter.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+This working group will track the discovery, production, and spread of solutions to challenges with InnerSource.
+We will track this production on a Kanban board with the following columns:
+
+1. **Challenge**.  This means that we've identified such-and-such a thing as a challenge.
+1. **Proposed**.  We have a proposal for a solution to the challenge.
+This could be to write an article, code up a tool, create a training video, etc.
+1. **Internal**.  We have implemented the solution internally in a company.
+1. **External**.  We have made the solution available externally.
+1. **Validated**.  More than one company is using the solution.
```
Comment (misappi) [inline ispo-working-group-charter.md:36]: I like this structured approach. The way it is decsribed here, reminds me of InnerSource patterns. Thus, I would expect that in many cases, these challenges together with the corresponding solutions would form (a) pattern(s).

File: ispo-working-group-charter.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+yet specific enough that people in the company can leverage them to solve their problems without needing notable amounts of company-specific work.
+
+# Examples
+
+As an example, to the challenge of `How do I sell InnerSource to Senior Leadership?`,
+this working group could ensure that there are video clips explaining various benefits of InnerSource from a Senior Leader perspective.
+There would be different clips that highlight different benefits of InnerSource,
+and the ISPO could select those videos that highlight the benefits that their Senior Leaders care about.
+
+As another example, to the challenge of `How do I show InnerSource KPIs?`,
```
Comment (misappi) [inline ispo-working-group-charter.md:20]: Just as a side remark wrt. KPI: Before it comes to the challenge of how I show InnerSource KPI's, IMO there is even a (pre-)challenge about how to select/define the relevant InnerSource KPI's.

Comment (rrrutledge) [reply ispo-working-group-charter.md:20]: Great point!  That one may come up for our working group, too!

File: ispo-working-group-charter.md (hunk 11 above)
Comment (rrrutledge) [reply ispo-working-group-charter.md:36]: Yes!  Suspect so!  A lot of the work here may end up producing patterns.  The working group will help us to organize ourselves sufficiently with the pre-work needed to crank out more patterns.

File: ispo-working-group-charter.md (hunk 10 above)
Comment (rrrutledge) [reply ispo-working-group-charter.md:24]: CamelCase word, I guess.  Not sure.  It's not terribly significant.

**Pull request merged**